pkg/crdpuller/discovery.go

```diff
@@ -495,7 +495,10 @@ func (sc *SchemaConverter) VisitKind(k *proto.Kind) {
 }
 
 func (sc *SchemaConverter) VisitReference(r proto.Reference) {
-	reference := r.Reference()
+	reference := r.Reference() // recursive CRDs are not supported
+	if reference == "io.k8s.apiextensions-apiserver.pkg.apis.apiextensions.v1.JSONSchemaProps" {
+		return
+	}
```

Comment:
what's the consequence of this? Who wants to crdpull CRDs?

Reply:
need to test crd pulling case :/ tbh this was very quick PR. Needs more testing. And this feedback kelps to identify the areas for testing :)

Reply:
Note that pulling **the** CRD type, not CRDs (= instances), is very meta 😄

Reply:
I think this does not impact CRD pulling. We already converts APIS into CRDs 
Tested crd-puller and it just worked. Might be some edge cases but not sure how to identify those without knowing what I'm looking for ;)